app/services/gemini_entity_enrichment_service.py

```
@@ -0,0 +1,668 @@
+            # Cache results
+            self._save_to_cache(cache_key, enhanced_results)
+            
+            logger.info(f"Successfully enriched {sum(len(entities) for entities in filtered_entities.values())} entities")
+            return enhanced_results
+            
+        except Exception as e:
+            logger.error(f"Entity enrichment failed: {e}")
+            logger.error(f"Exception type: {type(e).__name__}")
+            # Log more details if response is available
+            if 'response' in locals():
+                logger.error(f"Response status: {getattr(response, 'status_code', 'unknown')}")
```

Ruff (0.11.9)

Undefined name `response`

(F821)